src/wasm/wasm-validator.cpp

```diff
@@ -327,6 +327,57 @@ struct FunctionValidator : public WalkerPass<PostWalker<FunctionValidator>> {
         self->pushTask(visitPoppyExpression, currp);
       }
     }
+
+    // Also verify that only allowed expressions end up in the situation where
+    // the expression has type unreachable but there is no unreachable child.
+    // For example a Call with no unreachable child cannot be unreachable, but a
+    // Break can be.
+    if (curr->type == Type::unreachable) {
+      switch (curr->_id) {
+        case Expression::BreakId:
```

Comment:
If the break has a value, it should fall through to the general logic, right? (same with br_table)

Reply:
I don't think a value changes things here? The type is still unreachable even with a value. A condition can change the type, but we already have full validation logic for that in `visitBreak`. I added a comment clarifying all that now.

Reply:
Or right, I was thinking of `br_if` but forgot that it's a separate expression class. For br_table I was just confused.

Reply:
Oh wait, it's not a separate class. But the comment you added was exactly what I needed 👍